### PR TITLE
Literals fix

### PR DESCRIPTION
**Pull request by un-tone**
1) Replaced using of `First` property with `FirstOrDefault` method
2) Using `Db.SlowSQL` method instead of `Db.SQL`
3) Fixed string methods query building
4) Enumerable.Contains method handling

### CODE REVIEW

File: src/Visitors/WhereVisitor.cs [hunk 1]
```diff
@@ -148,24 +148,33 @@ public override void VisitIndex(IndexExpression node, QueryBuilder<TEntity> stat
 
         public override void VisitMethodCall(MethodCallExpression node, QueryBuilder<TEntity> state)
         {
+            ConstantExpression constNode = null;
             if (node.Method == KnownMethods.StringContains)
             {
                 state.WriteWhere("(");
                 Visit(node.Object, state);
+                constNode = node.Arguments[0] as ConstantExpression;
                 state.WriteWhere(" LIKE '%' || ? || '%')");
```
Comment (miyconst) [inline src/Visitors/WhereVisitor.cs]: I didn't know this actually works in Starcounter :O, but it does work.

File: src/QueryBuilder.cs [hunk 2]
```diff
@@ -110,9 +110,10 @@ public string BuildSqlString()
             return stringBuilder.ToString();
         }
 
-        public void AddVariable(object value) => Variables.Add(value);
+        public void AddVariable(object value) => VariableGetters.Add(() => value);
+        public void AddVariable(Func<object> valueGetter) => VariableGetters.Add(valueGetter);
 
-        public object[] GetVariables() => Variables.ToArray();
+        public object[] GetVariables() => VariableGetters.Select(x => x()).ToArray();
```
[the remark this answers is not in the document]
Comment (un-tone) [reply src/QueryBuilder.cs]: Fixed it

File: src/Visitors/WhereVisitor.cs [hunk 3]
```diff
@@ -133,12 +133,34 @@ public override void VisitMember(MemberExpression node, QueryBuilder<TEntity> st
...
+                var subNode = node;
+                while (subNode.Expression is MemberExpression memberNode)
+                {
+                    subNode = memberNode;
+                }
+                if (subNode.Expression is ConstantExpression)
+                {
+                    var objectMember = Expression.Convert(node, typeof(object));
+                    var getterLambda = Expression.Lambda<Func<object>>(objectMember);
+                    var getter = getterLambda.Compile();
```
[the remark this answers is not in the document]
Comment (un-tone) [reply src/Visitors/WhereVisitor.cs]: Thanks for the explanation. I fixed it.

**Comment by un-tone** [comment]
Please review the PR

**Pull request merged**